pkg/leech/storage.go

```diff
@@ -61,6 +61,7 @@ func (s *ObjectStore) Write(ctx context.Context, content io.Reader, objectDescri
 	writer := obj.NewWriter(ctx)
 
 	if _, err := io.Copy(writer, content); err != nil {
+		// Should we cancel context? obj.NewWriter() says to do so if you don't want ot write
```

Comment:
I don't see that in the docs, can you point me to a reference?

Reply:
https://pkg.go.dev/cloud.google.com/go/storage#ObjectHandle.NewWriter

>It is the caller's responsibility to call Close when writing is done. To stop writing without saving the data, cancel the context.

Reply:
Disappointingly, the example code neither closes nor cancels the context.